### PR TITLE
Adding CRP 2022-11-01 version

### PR DESCRIPTION
**Pull request by grizzlytheodore**
SDK PR for Swagger CRP 2022-11-01 version
https://github.com/Azure/azure-rest-api-specs/pull/21571

# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/IVirtualMachineScaleSetVMsOperations.cs [hunk 1]
```diff
@@ -54,7 +52,7 @@ public partial interface IVirtualMachineScaleSetVMsOperations
         /// <exception cref="Microsoft.Rest.ValidationException">
         /// Thrown when a required parameter is null
         /// </exception>
-        Task<AzureOperationResponse> ReimageWithHttpMessagesAsync(string resourceGroupName, string vmScaleSetName, string instanceId, bool? tempDisk = default(bool?), Dictionary<string, List<string>> customHeaders = null, CancellationToken cancellationToken = default(CancellationToken));
+        Task<AzureOperationResponse> ReimageWithHttpMessagesAsync(string resourceGroupName, string vmScaleSetName, string instanceId, VirtualMachineScaleSetVMReimageParameters vmScaleSetVMReimageInput = default(VirtualMachineScaleSetVMReimageParameters), Dictionary<string, List<string>> customHeaders = null, CancellationToken cancellationToken = default(CancellationToken));
```
Comment (grizzlytheodore) [inline sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/IVirtualMachineScaleSetVMsOperations.cs:55]: need customization or readme update

File: sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/IVirtualMachinesOperations.cs [hunk 2]
```diff
@@ -530,7 +534,7 @@ public partial interface IVirtualMachinesOperations
         /// <exception cref="Microsoft.Rest.ValidationException">
         /// Thrown when a required parameter is null
         /// </exception>
-        Task<AzureOperationResponse> ReimageWithHttpMessagesAsync(string resourceGroupName, string vmName, bool? tempDisk = default(bool?), Dictionary<string, List<string>> customHeaders = null, CancellationToken cancellationToken = default(CancellationToken));
+        Task<AzureOperationResponse> ReimageWithHttpMessagesAsync(string resourceGroupName, string vmName, VirtualMachineReimageParameters parameters = default(VirtualMachineReimageParameters), Dictionary<string, List<string>> customHeaders = null, CancellationToken cancellationToken = default(CancellationToken));
```
Comment (grizzlytheodore) [inline sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/IVirtualMachinesOperations.cs:537]: customization

File: sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/Models/VirtualMachineScaleSet.cs [hunk 3]
```diff
@@ -102,14 +102,16 @@ public VirtualMachineScaleSet()
...
...maticRepairsPolicy automaticRepairsPolicy = default(AutomaticRepairsPolicy), VirtualMachineScaleSetVMProfile virtualMachineProfile = default(VirtualMachineScaleSetVMProfile), string provisioningState = default(string), bool? overprovision = default(bool?), bool? doNotRunExtensionsOnOverprovisionedVMs = default(bool?), string uniqueId = default(string), bool? singlePlacementGroup = default(bool?), bool? zoneBalance = default(bool?), int? platformFaultDomainCount = default(int?), SubResource proximityPlacementGroup = default(SubResource), SubResource hostGroup = default(SubResource), AdditionalCapabilities additionalCapabilities = default(AdditionalCapabilities), ScaleInPolicy scaleInPolicy = default(ScaleInPolicy), string orchestrationMode = default(string), SpotRestorePolicy spotRestorePolicy = default(SpotRestorePolicy), PriorityMixPolicy priorityMixPolicy = default(PriorityMixPolicy), System.DateTime? timeCreated = default(System.DateTime?), bool? constrainedMaximumCapacity = default(bool?), VirtualMachineScaleSetIdentity identity = default(VirtualMachineScaleSetIdentity), IList<string> zones = default(IList<string>), ExtendedLocation extendedLocation = default(ExtendedLocation))
```
Comment (grizzlytheodore) [inline sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/Models/VirtualMachineScaleSet.cs:114]: customization

File: sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/Models/VirtualMachineScaleSetReimageParameters.cs [hunk 4]
```diff
@@ -36,12 +36,18 @@ public VirtualMachineScaleSetReimageParameters()
...
+        /// Disk.</param>
+        /// <param name="osProfile">Specifies information required for
+        /// reimaging the non-ephemeral OS disk.</param>
         /// <param name="instanceIds">The virtual machine scale set instance
         /// ids. Omitting the virtual machine scale set instance ids will
         /// result in the operation being performed on all virtual machines in
         /// the virtual machine scale set.</param>
-        public VirtualMachineScaleSetReimageParameters(bool? tempDisk = default(bool?), IList<string> instanceIds = default(IList<string>))
-            : base(tempDisk)
+        public VirtualMachineScaleSetReimageParameters(bool? tempDisk = default(bool?), string exactVersion = default(string), OSProfileProvisioningData osProfile = default(OSProfileProvisioningData), IList<string> instanceIds = default(IList<string>))
```
Comment (grizzlytheodore) [inline sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/Models/VirtualMachineScaleSetReimageParameters.cs:49]: customization

File: sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/VirtualMachineScaleSetVMsOperations.cs [hunk 5]
```diff
@@ -64,21 +64,19 @@ internal VirtualMachineScaleSetVMsOperations(ComputeManagementClient client)
...
+        /// Parameters for the Reimaging Virtual machine in ScaleSet.
         /// </param>
         /// <param name='customHeaders'>
         /// The headers that will be added to request.
         /// </param>
         /// <param name='cancellationToken'>
         /// The cancellation token.
         /// </param>
-        public async Task<AzureOperationResponse> ReimageWithHttpMessagesAsync(string resourceGroupName, string vmScaleSetName, string instanceId, bool? tempDisk = default(bool?), Dictionary<string, List<string>> customHeaders = null, CancellationToken cancellationToken = default(CancellationToken))
+        public async Task<AzureOperationResponse> ReimageWithHttpMessagesAsync(string resourceGroupName, string vmScaleSetName, string instanceId, VirtualMachineScaleSetVMReimageParameters vmScaleSetVMReimageInput = default(VirtualMachineScaleSetVMReimageParameters), Dictionary<string, List<string>> customHeaders = null, CancellationToken cancellationToken = default(CancellationToken))
```
Comment (grizzlytheodore) [inline sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/VirtualMachineScaleSetVMsOperations.cs:76]: customization for readme

File: sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/VirtualMachineScaleSetVMsOperationsExtensions.cs [hunk 6]
```diff
@@ -38,14 +38,12 @@ public static partial class VirtualMachineScaleSetVMsOperationsExtensions
...
             /// The instance ID of the virtual machine.
             /// </param>
-            /// <param name='tempDisk'>
-            /// Specifies whether to reimage temp disk. Default value: false. Note: This
-            /// temp disk reimage parameter is only supported for VM/VMSS with Ephemeral OS
-            /// disk.
+            /// <param name='vmScaleSetVMReimageInput'>
+            /// Parameters for the Reimaging Virtual machine in ScaleSet.
             /// </param>
-            public static void Reimage(this IVirtualMachineScaleSetVMsOperations operations, string resourceGroupName, string vmScaleSetName, string instanceId, bool? tempDisk = default(bool?))
```
Comment (grizzlytheodore) [inline sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/VirtualMachineScaleSetVMsOperationsExtensions.cs:46]: customization or readme

File: sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/Models/VirtualMachineScaleSet.cs (hunk 3 above)
Comment (grizzlytheodore) [reply sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/Models/VirtualMachineScaleSet.cs:114]: done

**Review by archerzz — COMMENTED** [review]
@grizzlytheodore Is this just a code update or you want to publish a new release? Note that you need to update `AssemblyInfo.cs` if you want to publish a new release.

**Comment by grizzlytheodore** [comment]
Hello @archerzz, assemblyInfo is not being changed in this PR, because we had a[ SDK PR for 2022-09-04 CloudService release](https://github.com/Azure/azure-sdk-for-net/pull/32626) merged a couple weeks ago which [updates the version to 60.0.0, ](https://github.com/Azure/azure-sdk-for-net/pull/32626/files#diff-83ec468f41c9e36989f0c5750d54d930e6b37acb6d27b1ab083446a6dc8ff4a9)but we did not publish that version because we wanted to publish with this new CRP version. 

[you can see here 
](https://www.nuget.org/packages/Microsoft.Azure.Management.Compute) the latest version published is 59.0.0 and the main branch has 60.0.0 

please merge !

**Review by archerzz — APPROVED** [review]

**Comment by archerzz** [comment]
> Hello @archerzz, assemblyInfo is not being changed in this PR, because we had a[ SDK PR for 2022-09-04 CloudService release](https://github.com/Azure/azure-sdk-for-net/pull/32626) merged a couple weeks ago which [updates the version to 60.0.0, ](https://github.com/Azure/azure-sdk-for-net/pull/32626/files#diff-83ec468f41c9e36989f0c5750d54d930e6b37acb6d27b1ab083446a6dc8ff4a9)but we did not publish that version because we wanted to publish with this new CRP version.
> 
> [you can see here ](https://www.nuget.org/packages/Microsoft.Azure.Management.Compute) the latest version published is 59.0.0 and the main branch has 60.0.0
> 
> please merge !

Thanks. Merging

**Pull request merged**